.github/workflows/tests.yml

```diff
@@ -2,43 +2,46 @@ name: Tests
 
 on:
   push:
-    branches:
-      - master
+    branches: [master]
   pull_request:
 
 jobs:
   unit-tests:
+    strategy:
+      matrix:
+        os: [ 'ubuntu-latest' ]
+        ruby: [ '2.5.5' ]
+        bundler: [ '1.17.3' ]
+    runs-on: ${{ matrix.os }}
     name: Unit tests
-    runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout
+      - name: Checkout ${{ github.sha	}}
         uses: actions/checkout@v2
-      - name: Setup Ruby
-        uses: actions/setup-ruby@v1
-        with:
-          ruby-version: '2.5'
-      - name: Cache dependencies
-        uses: actions/cache@v2
+
+      - name: Setup Ruby ${{ matrix.ruby }} and Bundler ${{ matrix.bundler }}
+        uses: ruby/setup-ruby@v1
         with:
-          path: ~/vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('apps/*/Gemfile.lock') }}
```

Comment:
Why get rid of the cache action? You can see in the runs they have to re-install the gems every time.

Reply:
`ruby/setup-ruby@v1` https://github.com/ruby/setup-ruby provides caching out of the box with the `bundler-cache` key:

```yml
steps:
  - name: Setup Ruby ${{ matrix.ruby }} and Bundler ${{ matrix.bundler }}
    uses: ruby/setup-ruby@v1
    with:
      ruby-version: ${{ matrix.ruby }}
      bundler-cache: true
      bundler: ${{ matrix.bundler }}
```

Reply:
I'm not sure if it keeps the cache from run to run. You can see here in the most recent run that it's fetching and installing. Should read 'using'. Plus you can tell by the runtimes. It's 5 minutes here when it's only 1 on master.  Am I missing something?

Reply:
It's because there is no Gemfile at root of repo. I looked at the ruby action code and some pull requests that would build cache hash off Gemfile.lock and fall back to Gemfile, but at root of repo.